sql/sys_vars.inl

```diff
@@ -795,6 +795,51 @@ protected:
   bool set_filter_value(const char *value, Master_info *mi);
 };
 
+class Sys_var_binlog_filter: public sys_var
+{
+private:
+  int opt_id;
+  privilege_t m_access_global;
+
+public:
+  Sys_var_binlog_filter(const char *name, int getopt_id, const char *comment,
+                     privilege_t access_global)
+    : sys_var(&all_sys_vars, name, comment, sys_var::GLOBAL, 0, NO_GETOPT,
+              NO_ARG, SHOW_CHAR, 0, NULL, VARIABLE_NOT_IN_BINLOG,
+              NULL, NULL, NULL), opt_id(getopt_id),
+      m_access_global(access_global)
+  {
+    option.var_type|= GET_STR | GET_ASK_ADDR;
+  }
+
+  bool do_check(THD *thd, set_var *var) override
+  {
+    return Sys_var_charptr::do_string_check(thd, var, charset(thd));
```

Comment:
As the variable is read-only, we don't need `do_check()` either, that can also just be (I missed it in my mention initially, sorry)
```
  {
    DBUG_ASSERT(FALSE);
    return true;
  }
```